### PR TITLE
[TEST] Migrated REST tests extending JsonLogsIntegTestCase

### PR DESCRIPTION
**Pull request by jozala**
REST tests extending JsonLogsIntegTestCase migrated to the new REST testing framework, using 'elasticsearch.internal-java-rest-test' Gradle plugin

### CODE REVIEW

File: qa/unconfigured-node-name/src/javaRestTest/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java [hunk 1]
```diff
@@ -30,6 +31,30 @@ public class JsonLogsFormatAndParseIT extends JsonLogsIntegTestCase {
     private static final String COMPUTERNAME = "WindowsComputername";
     private static final String HOSTNAME = "LinuxDarwinHostname";
 
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .setting("xpack.security.enabled", "false")
+        .withNode(localNodeSpecBuilder -> localNodeSpecBuilder.name(OS_NAME.startsWith("Windows") ? COMPUTERNAME : HOSTNAME))
+        // TODO @jozala -
+        .settingsModifier(
```
Comment (jozala) [inline qa/unconfigured-node-name/src/javaRestTest/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java:39]: I've added this settings modifier to be able to remove `node.name` from configuration, but keep the rest of the Elasticsearch node setup working. I am wondering if there is already a better way to achieve this.

File: distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java [hunk 2]
```diff
@@ -10,36 +10,45 @@
...
+
+    @Override
+    protected String getLogFileName() {
+        // TODO @jozala REFACTOR to be removed
+        return "server.log";
     }
 
     @Override
     protected BufferedReader openReader(Path logFile) {
-        assumeFalse(
```
Comment (jozala) [inline distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java:33]: I haven't recreated this behaviour. Do we run these tests against external clusters? Do we need this?

File: distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java [hunk 3]
```diff
@@ -10,36 +10,45 @@
...
-            "Skipping test because it is being run against an external cluster.",
-            logFile.getFileName().toString().equals("--external--")
-        );
-        return AccessController.doPrivileged((PrivilegedAction<BufferedReader>) () -> {
-            try {
-                return Files.newBufferedReader(logFile, StandardCharsets.UTF_8);
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        });
```
Comment (jozala) [inline distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java:43]: I haven't recreated this behaviour. Do we run these tests against external clusters? Do we need this?

File: qa/unconfigured-node-name/src/javaRestTest/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java [hunk 4]
```diff
@@ -30,6 +31,30 @@ public class JsonLogsFormatAndParseIT extends JsonLogsIntegTestCase {
...
+            settings -> settings.entrySet()
+                .stream()
+                .filter(it -> it.getKey().equals("node.name") == false)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+        )
```
Comment (jozala) [inline qa/unconfigured-node-name/src/javaRestTest/java/org/elasticsearch/unconfigured_node_name/JsonLogsFormatAndParseIT.java:44]: I've added this settings modifier in configuration to be able to remove `node.name` from configuration, but keep the rest of the Elasticsearch node setup working. I am wondering if there is already a better way to achieve this.

File: distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java (hunk 3 above)
Comment (jozala) [reply distribution/archives/integ-test-zip/src/javaRestTest/java/org/elasticsearch/test/rest/JsonLogsFormatAndParseIT.java:43]: Discussed with @breskeby and believe we do not run these tests against external clusters.
In theory when running the tests against external cluster some tests would be ignored using this code.
I have removed this for now and I will get back to it if see some tests are failing.

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/DefaultSettingsProvider.java [hunk 5]
```diff
@@ -70,17 +76,20 @@ public Map<String, String> get(LocalNodeSpec nodeSpec) {
...
         }
 
-        settings.put("cluster.initial_master_nodes", "[" + masterEligibleNodes + "]");
-        settings.put("discovery.seed_providers", "file");
-        settings.put("discovery.seed_hosts", "[]");
+        if (masterEligibleNodes.isEmpty() == false) {
+            settings.put("cluster.initial_master_nodes", "[" + masterEligibleNodes + "]");
+            settings.put("discovery.seed_providers", "file");
+            settings.put("discovery.seed_hosts", "[]");
+        }
```
Comment (jozala) [inline test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/DefaultSettingsProvider.java:92]: @mark-vieira @breskeby 
This is something I would definitely like your opinion on. I am now allowing creating cluster without setting the `cluster.initial_master_nodes` where it was explicitly disallowed previously. I believe this is OK when there is a single node, but it possible can create some problems in more complex tests. 
I need that to be able to start the node without the explicit name in the `JsonLogsFormatAndParseIT` test.

Comment (mark-vieira) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/DefaultSettingsProvider.java:92]: Well, in the case where the node doesn't have a name, `masterEligibleNodes` here would be an empty string, no? Would setting `initial_master_nodes` to `[]` be equivalent to not setting it at all or is the behavior different?

Comment (mark-vieira) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/DefaultSettingsProvider.java:92]: Alternatively, could we make that above error condition more specific. That is, only throw an exception when there are no master eligible nodes _and_ the cluster contains multiple nodes?

File: x-pack/plugin/ent-search/qa/full-cluster-restart/build.gradle [hunk 6]
```diff
@@ -13,7 +13,6 @@ apply plugin: 'elasticsearch.bwc-test'
 
 
 dependencies {
-  testImplementation project(path: xpackModule('ent-search'))
```
Comment (breskeby) [inline x-pack/plugin/ent-search/qa/full-cluster-restart/build.gradle:16]: what's this about? not needed?

**Review by breskeby — APPROVED** [review]
I think this looks good. I left one remark for double checking for cleanup but overall this is good to merge.

With having the struggle with the variant selection, I think the RestTestBasePlugin looks also better now being more idiomatic to what a base plugin in gradle terms would do, meaning just provide infrastructure to the build but don't be opinionated or add certain assumptions. 

Also having less logic in our build scripts makes things way easier to maintain for the team. 👍

**Pull request merged**